### PR TITLE
Bug 1882775: Prevent creation of duplicate PVs for same device

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1882775

### CODE REVIEW

File: pkg/diskmaker/diskmaker.go [hunk 1]
```diff
@@ -310,3 +341,95 @@ func fileExists(filename string) bool {
 	}
 	return true
 }
+
+// getPVCreationLock checks whether a PV can be created based on this device
+// and Locks the device so that no PVs can be created on it while the lock is held.
+// the PV lock will fail if:
+// - another process holds an exclusive file lock on the device (using the syscall flock)
+// - a symlink to this device exists in symlinkDirs
+// returns:
```
Comment (huffmanca) [inline pkg/diskmaker/diskmaker.go:350]: This list of returns doesn't seem to be correct any longer.

**Review by huffmanca — COMMENTED** [review]
Just one nitpick involving comments. I've reviewed the logic, and this makes sense to me.

File: pkg/diskmaker/diskmaker.go (hunk 1 above)
Comment (gnufied) [reply pkg/diskmaker/diskmaker.go:350]: fixed

**Pull request merged**